tests/core/providers/test_provider.py

```diff
@@ -51,3 +52,18 @@ def must_not_call():
     assert w3.isConnected()
 
     assert isinstance(auto._active_provider, ConnectedProvider)
+
+
+def test_encode_rpc_request():
+    import pytest
+    import datetime
+
+    provider = JSONBaseProvider()
+
+    method = 'test_method'
+    params = []
+    expected_bstr = b'{"jsonrpc": "2.0", "method": "test_method", "params": [], "id": 0}'
+    assert provider.encode_rpc_request(method, params) == expected_bstr
+    params = [datetime.datetime.utcnow()]
+    with pytest.raises(TypeError, match=r'.* failure .*'):
```

Comment:
Maybe match against `datetime` to show that the repr of the field we care about was included?